src/main/java/org/openstreetmap/atlas/generator/AtlasGenerator.java

```diff
@@ -534,6 +675,242 @@ protected SwitchList switches()
         return super.switches().with(COUNTRIES, COUNTRY_SHAPES, SHARDING_TYPE, PBF_PATH, PBF_SCHEME,
                 PBF_SHARDING, PREVIOUS_OUTPUT_FOR_DELTA, CODE_VERSION, DATA_VERSION,
                 EDGE_CONFIGURATION, WAY_SECTIONING_CONFIGURATION, PBF_NODE_CONFIGURATION,
-                PBF_WAY_CONFIGURATION, PBF_RELATION_CONFIGURATION, ATLAS_SCHEME);
+                PBF_WAY_CONFIGURATION, PBF_RELATION_CONFIGURATION, ATLAS_SCHEME, USE_RAW_ATLAS);
+    }
+
+    /**
+     * @param sparkContext
+     *            Spark context (or configuration) as a key-value map
+     * @param previousOutputForDelta
+     *            Previous Atlas generation delta output location
+     * @return A Spark {@link PairFlatMapFunction} that takes a tuple of a country shard name and
+     *         atlas file and returns all the {@link AtlasDelta} for the country
+     */
+    private PairFlatMapFunction<Tuple2<String, Atlas>, String, AtlasDelta> computeAtlasDelta(
```

Comment:
Make all those functions static so the whole SparkJob does not have to be serialized and passed around for the slaves to be able to call those methods!

Reply:
And even better: push them to the helper!

Reply:
Good call, let me do this!